src/ansi.rs

```diff
@@ -672,6 +673,70 @@ pub trait Handler {
 
     /// Set mouse cursor icon.
     fn set_mouse_cursor_icon(&mut self, _: CursorIcon) {}
+
+    /// Report current keyboard mode.
+    ///
+    /// ## Caveat
+    ///
+    /// This request must be done successively with the [`identify_terminal`] request, given
+    /// clients rely on the later to identify whether the terminal suppors kitty's keyboard
+    /// protocol.
+    ///
+    /// [`identify_terminal`]: Self::identify_terminal
+    fn report_keyboard_mode(&mut self) {}
+
+    /// Push keyboard mode into the keyboard mode stack.
+    fn push_keyboard_mode(&mut self, _mode: KeyboardMode) {}
+
+    /// Pop the given amount of keyboard modes from the
+    /// keyboard mode stack.
+    fn pop_num_keyboard_modes(&mut self, _to_pop: u16) {}
+
+    /// Set the [`keyboard mode`] using the given [`behavior`].
+    ///
+    /// [`keyboard mode`]: crate::ansi::KeyboardMode
+    /// [`behavior`]: crate::ansi::KeyboardModeApplyBehavior
+    fn set_keyboard_mode(&mut self, _mode: KeyboardMode, _behavior: KeyboardModeApplyBehavior) {}
+}
+
+bitflags! {
+    /// The keyboard mode representing [`kitty keyboard protocol'].
+    ///
+    /// The details on how to build the escape sequences could be found in the
+    /// mentioned specification.
+    ///
+    /// [`kitty keyboard protocol']: https://sw.kovidgoyal.net/kitty/keyboard-protocol
+    #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    pub struct KeyboardMode : u8 {
+        /// No particular mode is being set.
+        const NO_MODE                 = 0b0000_0000;
+        /// Report `Esc`, `alt` + `key`, `ctrl` + `key`, `ctrl` + `alt` + `key`, `shift`
+        /// + `alt` + `key` keys using `CSI u` sequence instead of legacy ones.
```

Comment:
```suggestion
        /// Report `Esc`, `alt` + `key`, `ctrl` + `key`, `ctrl` + `alt` + `key`, `shift`
        /// + `alt` + `key` keys using `CSI u` sequence instead of raw ones.
```

Not using this protocol isn't really "legacy". It is and always will be one of the easiest and most convenient ways to handle text input for terminal applications.

I've picked "raw" here because it was the first thing that came to my mind, but I'm open for alternative suggestions.